### ISSUE TITLE
Generic size types

### ISSUE THREAD

**Issue by tversteeg** [kornelski/imgref#24]
Would you be willing to accept a PR where I make each numeric type generic? Some crates use `u32` for sizes, some `usize` and some even `u16`.

**Comment by kornelski** [comment]
Yes. Please see https://github.com/kornelski/imgref/tree/v2-preview

**Comment by tversteeg** [comment]
Oh great! Do you need any help with this new version?

**Comment by kornelski** [comment]
Yes, help is welcome. 

That code is unfinished and buggy. I wish more of it could be implemented using safe code, even as an option, but I haven't figured out how to structure the traits for it.
